pkg/proc/proc.go

```diff
@@ -543,9 +551,12 @@ func GoroutinesInfo(dbp Process) ([]*G, error) {
 			allg = append(allg, g)
 		}
 	}
-	dbp.Common().allGCache = allg
+	// Save the set for later only if we've proceessed all goroutines
+	if start == 0 && nextg == 0 {
```

Comment:
This will never cache anything because nextg now is never 0. It's probably better to return directly in the `if count != 0 && len(allg) >= count` and cache unconditionally here. I think.